library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -1169,9 +1146,10 @@ public void setWarmGps(boolean warmGps, @Nullable String rejectionDialogTitle, @
             hasAlreadyRequestedEnableLocation = false;
             // In case the location settings were turned off while the warmGps is still true, it means that the LocationClient is also on
             // We should just re-enable the location so that the LocationClient can reconnect to the location services
-            checkLocationSettingsAndStartLocationServices(shouldStartNow);
+            checkLocationSettingsAndStartLocationServices(shouldStartNow, onLocationServicesEnabledCallBack);
+        } else if (onLocationServicesEnabledCallBack != null) {
+            onLocationServicesEnabledCallBack.onSuccess();
```

Comment:
Should this be called here?

Reply:
Yes, because location services are already enabled.

Reply:
This is called if `warmGps` is not true. Isn't that a bug

Reply:
True, I thought the check was for `shouldStartNow`.